### PR TITLE
feat(bulk_create): support sending the qty and location_type to bulk …

### PR DESCRIPTION
**Pull request by maurogri-th**
…create stock on product bulk creation

### CODE REVIEW

File: lib/v1/products/base.js [hunk 1]
```diff
@@ -24,6 +24,27 @@ const stockConfigurationLocation = {
...
+              'warehouse',
+              'warehouse_shelf'
+            ]
+          },
+          {
+            type: 'null'
+          }
+        ]
+      },
+      qty: oneOf({
```
Comment (EvgeniyStrigo) [inline lib/v1/products/base.js:45]: oneOf of only one variant? ) It seems strange )

**Review by EvgeniyStrigo — CHANGES_REQUESTED** [review]

**Comment by erezsob** [comment]
I don't know if it makes sense to add fields to the schema for all the products' requests just for a very specific scenario. It means if the product handler will be called with these properties (since now it's allowed per the schema changes), it will pass schema validation but fail against the DB.
Maybe might make more sense to create a new schema specifically for bulk_create that will inherit and extend the products' schema?

File: lib/v1/products/base.js (hunk 1 above)
Comment (maurogri-th) [reply lib/v1/products/base.js:45]: oneOf has inside the option of null

**Comment by maurogri-th** [comment]
> I don't know if it makes sense to add fields to the schema for all the products' requests just for a very specific scenario. It means if the product handler will be called with these properties (since now it's allowed per the schema changes), it will pass schema validation but fail against the DB.
> Maybe might make more sense to create a new schema specifically for bulk_create that will inherit and extend the products' schema?

Thats true, it would still fail if passed (unless i change it in the API), but at the same time makes sense that it should be there. Also would make sense that the creation of product with stock should be handled with one call.

File: lib/v1/products/base.js (hunk 1 above)
Comment (EvgeniyStrigo) [reply lib/v1/products/base.js:45]: Ok

**Review by EvgeniyStrigo — APPROVED** [review]

**Comment by maurogri-th** [comment]
As we are gonna support the stock creation inside the product creation, we will keep this schema change.

**Pull request merged**